### PR TITLE
Clean up INoteBookEditor & related interfaces

### PR DESCRIPTION
**Pull request by DonJayamanne**
More cleanup as part of removing WebView code (lots of work remaining, this is just one of the many PRs will have).

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
             "smartStep": true,
             "sourceMaps": true,
             "outFiles": ["${workspaceFolder}/out/**/*", "!${workspaceFolder}/**/node_modules**/*"],
-            "preLaunchTask": "Compile",
+            // "preLaunchTask": "Compile",
```
Comment (rchiodo) [inline .vscode/launch.json:15]: I assume you don't want to submit this.

Comment (DonJayamanne) [reply .vscode/launch.json:15]: First Ian, now you... why can't i check in crap...

Comment (DonJayamanne) [reply .vscode/launch.json:15]: (pun intended)

File: src/client/datascience/notebook/notebookEditorProviderWrapper.ts [hunk 2]
```diff
@@ -25,9 +25,6 @@ export class NotebookEditorProviderWrapper implements INotebookEditorProvider {
     public get onDidCloseNotebookEditor(): Event<INotebookEditor> {
```
Comment (rchiodo) [inline src/client/datascience/notebook/notebookEditorProviderWrapper.ts:25]: Do we even need this class anymore?

File: .vscode/launch.json (hunk 1 above)
Comment (DonJayamanne) [reply .vscode/launch.json]: Time for me to create a new launch config

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/dataScienceSurveyBanner.ts [hunk 3]
```diff
@@ -122,7 +121,6 @@ export class DataScienceSurveyBanner implements IJupyterExtensionBanner, IExtens
         this.setPersistentState(BannerType.DSSurvey, DSSurveyStateKeys.ShowBanner);
```
Comment (IanMatthewHuff) [inline src/client/datascience/dataScienceSurveyBanner.ts:121]: We could pull the DSSurvey banner as well I think. Believe that is webviews only.

Comment (DonJayamanne) [reply src/client/datascience/dataScienceSurveyBanner.ts:121]: I've left that code as is, didn't want to chagne that.
I think we have a seperate issue for banners/surveys in native notebook. hence left this.

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

File: src/client/datascience/notebook/notebookEditorProviderWrapper.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/datascience/notebook/notebookEditorProviderWrapper.ts:25]: Fixed

**Pull request merged**